lib/waistband/index.rb

```diff
@@ -314,6 +349,24 @@ def base_config_name
         "#{@index_name}_#{::Waistband.config.env}"
       end
 
+      def check_permission!(permission)
+        raise "::Waistband::Errors::Permissions::#{permission.classify}".constantize.new("Don't have enough permissions to #{permission}") unless check_permission?(permission)
```

Comment:
Errors about permissions are always tricky to write.  I think perhaps something like this would be better:
`"#{self.class} does not have the required permission, #{permission}, for the #{operation_name} operation"`

This would require this method signature to change to `def check_permission!(permission, operation_name)`